src/main/java/TicTacToe.java

```diff
@@ -0,0 +1,327 @@
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+// Strategy pattern.
+// The computer player's behavior/strategy can be replaced by inheriting from the interface below
+// Also, the human player's behavior inherits from the same interface
+// This also makes it easy to modify the game for 2 human players, 2 computer players etc.
+
+interface MoveMethod
+{
+    public int move();
+}
+
+class SimpleMoveStrategy implements MoveMethod
+{
+    private TicTacToe game;
+    public SimpleMoveStrategy(TicTacToe t) {
+        game = t ;
+    }
+
+    public int move() {
+
+        for( int i = 0 ; i < TicTacToe.N ; i ++ )
+            {
+                for( int j = 0 ; j < TicTacToe.N ; j ++ )
+                    {
+                        if( game.board[i][j] == 0 )
+                            return (i*3+j+1) ;
+                    }
+            }
+        return 0 ;

```

Comment:
This strategy works BUT if the human player plays the same moves, you get the same game every time.
Also, this strategy is naive to just go in the first open spot every time.

Could do a simple randomization to place the pieces to make for a more exciting game.